src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java

```diff
@@ -1,54 +1,179 @@
 import org.moltimate.moltimatebackend.model.Motif;
 import org.moltimate.moltimatebackend.model.Residue;
-import org.moltimate.moltimatebackend.validation.exceptions.InvalidFileException;
-import org.moltimate.moltimatebackend.validation.exceptions.InvalidPdbIdException;
 import org.springframework.core.io.ByteArrayResource;
 import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.multipart.MultipartFile;
 
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FileUtils {
 
     private static final PDBFileReader PDB_FILE_READER = new PDBFileReader();
     private static final MMCIFFileReader MMCIF_FILE_READER = new MMCIFFileReader();
+    private static final String MOTIF_DATA_SEPARATOR = "MOLTIMATE-DATA\n";
 
-    public static Motif getMotifFromFile(MultipartFile file) {
-        // TODO: Read motif metadata from file and process it
-        return null;
+    private static enum ProteinFileType {
+        PDB(".pdb"),
+        MMCIF(".cif"),
+        MOTIF(".motif");
+
+        private String fileExtension;
+
+        ProteinFileType(String fileExtension) {
+            this.fileExtension = fileExtension;
+        }
+
+        @Override
+        public String toString() {
+            return fileExtension;
+        }
+
+        public static List<String> getValidFileTypes() {
+            return Arrays.stream(ProteinFileType.values())
+                    .map(proteinFileType -> proteinFileType.toString())
+                    .collect(Collectors.toList());
+        }
+    }
+
+    public static ResponseEntity<Resource> createPdbFile(String pdbId) {
+        Resource pdbFile = new ByteArrayResource(ProteinUtils.queryPdb(pdbId)
+                                                         .toPDB()
+                                                         .getBytes());
+        return createResponseFile(pdbFile, pdbId, ProteinFileType.PDB);
+    }
+
+    public static ResponseEntity<Resource> createMmcifFile(String pdbId) {
+        Resource mmcifFile = new ByteArrayResource(ProteinUtils.queryPdb(pdbId)
+                                                           .toMMCIF()
+                                                           .getBytes());
+        return createResponseFile(mmcifFile, pdbId, ProteinFileType.MMCIF);
+    }
+
+    public static ResponseEntity<Resource> createMotifFile(MakeMotifRequest request) {
+        return createMotifFile(request.getPdbId(), request.getEcNumber(), request.getActiveSiteResidues());
+    }
+
+    public static ResponseEntity<Resource> createMotifFile(String pdbId, String ecNumber, List<Residue> activeSiteResidues) {
+        String pdbFile = ProteinUtils.queryPdb(pdbId).toPDB();
+        List<String> residueStrings = activeSiteResidues.stream()
```

Comment:
can we have this method call the below method (since we can get the structure from the pdb and then call the below)

Reply:
Done